### PR TITLE
[.NET] Arabic Duration support

### PR DESCRIPTION
**Pull request by LionbridgeCSII**
All test cases pass extraction.
26 cases pass resolution.

Some of the duplicate test cases that were previously removed have been reintroduced with variations.

### CODE REVIEW

File: Specs/DateTime/Arabic/DurationExtractor.json [hunk 1]
```diff
@@ -222,20 +289,18 @@
...
   {
     "Input": "سأغادر ليوم كامل",
-    "NotSupported": "dotnet",
     "NotSupportedByDesign": "java, javascript, python",
     "Results": [
       {
-        "Text": " ليوم كامل",
-        "Start": 6,
-        "Length": 10,
+        "Text": "يوم كامل",
```
Comment (tellarin) [inline Specs/DateTime/Arabic/DurationExtractor.json:295]: Why isn't the "ل" tagged here?

**Review by tellarin — APPROVED** [review]
LGTM. But please reply to that one comment. Thanks.

**Comment by tellarin** [comment]
@LionbridgeCSII, could we get a doc report explaining the edits in the spec cases? At least for the majority of cases?
This may help in the future with the language expansion process.

File: Specs/DateTime/Arabic/DurationExtractor.json (hunk 1 above)
Comment (LionbridgeCSII) [reply Specs/DateTime/Arabic/DurationExtractor.json:295]: The original extraction " ليوم كامل" means "for the whole day", the updated one "يوم كامل" only contains "whole day". 
In English the same happens (for the input "for the whole day", only "whole day" is extracted).

Comment (tellarin) [reply Specs/DateTime/Arabic/DurationExtractor.json:295]: I see. But in this case, if we don't apply a tokenizer than the enity stops in the middle of a word, so it wouldn't be a problem to include the "ل".
Are there many other similar cases in Arabic?

Comment (LionbridgeCSII) [reply Specs/DateTime/Arabic/DurationExtractor.json:295]: yes it is common in Arabic, the main particles that can be linked to other words (in DateTime context) are "ل" (for/at), "ب" (in) and  "و" (and). For example: 
"لسابعة" (at seven) (ل+سابعة)
"بالشهر" (in the month) (ب+الشهر)
"الإثنين   والثلاثاء" (Monday and Tuesday) (و+الثلاثاء)

**Comment by LionbridgeCSII** [comment]
@tellarin, here are the spec changes:

Changes not due to duplicate cases (appearing in both Extractor and Parser):
- "سأغادر لمدة سنة كامل" (Grammatical error) -> "سأغادر لمدة سنة كاملة" (I'll leave for full-year)
- "سأغادر لمدة عام واحد وشهر و 21 يومًا" (Removed Arabic diacritics***) -> "سأغادر لمدة عام واحد و شهر و 21 يوما" (I'll leave for 1 year 1 month 21 days)
- "سأغادر لمدة يومين وشهر واحد" (Modified to better match the English version and include digit numbers combined with written numbers and units) -> "سأغادر لمدة 2 أيام و شهر واحد" (I'll leave for 2 days 1 month)
- " 3 دقائق" (removed trailing space) -> "3 دقائق" (3 minutes)

*** Regarding the use of diacritics in Arabic, the best solution would be to fully support them, however, since a word can have many diacritical variations, that would require very large and complex regexes. On the other hand, in practice diacritics are used mostly to disambiguate ambiguous words and this does not seem to be case for DateTime entities. So, our approach is to discard diacritics for the time being and possibly add support for them in the future if it turns out to be necessary.

Changes due to duplicate cases:
DurationExtractor:
- "سأترك لمدة 3 ساعات" (duplicate) -> "سأترك مدة 3 ساعات" (I'll leave for 3h)
- "سأغادر لمدة 3 ساعات" (duplicate) -> "سأغادر لمدة 3 الساعات" (I'll leave for 3 hr)
- "سأغادر لمدة 3 ساعات" (duplicate) -> "سأغادر لمدة 3 ساعة" (I'll leave for 3 hr)
- "سأغادر لمدة 3 أيام" (duplicate) -> "سأغادر مدة 3 أيام" (I'll leave for 3day)
- "سأترك لمدة 3 دقائق" (duplicate) -> "سأترك مدة 3 دقائق" (I'll leave for 3-min)
- "سأغادر طوال اليوم" (duplicate) -> "سأغادر طوال يوم" (I'll leave for all day)
- "سأغادر طوال الأسبوع" (duplicate) -> "سأغادر طوال أسبوع" (I'll leave for all week)
- "سأغادر طوال الشهر" (duplicate) -> "سأغادر طوال شهر" (I'll leave for all month)
- "سأغادر ليوم كامل" (duplicate) -> "سأغادر يوم كامل" (I'll leave for full day)
- "سأغادر لمدة أسبوع كامل" (duplicate) -> "سأغادرلمدة أسبوع كامل" (I'll leave for full week)
- "سأغادر لمدة شهر كامل" (duplicate) -> "سأغادرلمدة شهر كامل" (I'll leave for full month)
- "سأغادر لمدة نصف ساعة" (duplicate) -> "سأغادر مدة نصف ساعة" (I'll leave for a half hour)
- "سأغادر لمدة ساعة ونصف" (duplicate) -> "سأغادرمدة ساعة ونصف" (I'll leave for an hour and half)
- "سأغادر لمدة نصف ساعة" (duplicate) -> "سأغادر لمدة نصف الساعة" (I'll leave for a half hour)

DurationParser:
- "سأترك لمدة 3 ساعات" (duplicate) -> "سأترك مدة 3 ساعات" (I'll leave for 3h)
- "سأغادر لمدة 3 ساعات" (duplicate) -> "سأغادر لمدة 3 الساعات" (I'll leave for 3 hr)
- "سأغادر لمدة 3 ساعات" (duplicate) -> "سأغادر لمدة 3 ساعة" (I'll leave for 3 hr)
- "سأغادر لمدة 3 ساعات" (duplicate) -> "سأغادر مدة 3 أيام" (I'll leave for 3 hr)
- "سأغادر لمدة 3 أيام" (duplicate) -> removed (I'll leave for 3day)
- "سأغادر طوال اليوم" (duplicate) -> "سأغادر طوال يوم" (I'll leave for all day)
- "سأغادر طوال الأسبوع" (duplicate) -> "سأغادر طوال أسبوع" (I'll leave for all week)
- "سأغادر طوال الشهر" (duplicate) -> "سأغادر طوال شهر" (I'll leave for all month)
- "سأغادر ليوم كامل" (duplicate) -> "سأغادر يوم كامل" (I'll leave for full day)
- "سأغادر لمدة أسبوع كامل" (duplicate) -> "سأغادر مدة أسبوع كامل" (I'll leave for full week)
- "سأغادر لمدة شهر كامل" (duplicate) -> "سأغادر مدة شهر كامل" (I'll leave for full month)
- "سأغادر طوال اليوم" (duplicate) -> removed (I'll leave for all day)
- "سأغادر لمدة ساعة ونصف" (duplicate) -> "سأغادر مدة ساعة ونصف" (I'll leave for an hour and half)

**Pull request merged**